packages/shared/src/net/retry.ts

```diff
@@ -44,14 +46,19 @@ export async function executeWithNetworkRetries<T>(
   options?: Partial<ExecuteWithHttpRetriesOptions>
 ): Promise<T> {
   const actualOptions = { ...defaultOptions, ...options };
-  const { httpCodesToRetry } = actualOptions;
+  const { httpCodesToRetry, httpStatusCodesToRetry } = actualOptions;
   const codesAsString = httpCodesToRetry.map(String);
   return executeWithRetries(fn, {
     ...actualOptions,
     shouldRetry: (_, error: unknown) => {
       const networkCode = axios.isAxiosError(error) ? error.code : undefined;
-      if (!networkCode) return false;
-      return codesAsString.includes(networkCode);
+      const networkStatus = axios.isAxiosError(error) ? error.response?.status : undefined;
+      if (!networkCode && !networkStatus) return false;
```

Comment:
NABD, but since we have a fallback on the return statement, could prob remove this right?